lib/assessment.rb

```diff
@@ -5,35 +5,45 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
-##
-# your code here
-##
+class Dog
+  attr_reader :name, :breed, :age
+
+  def initialize name, breed, age
+    @name, @breed, @age = name, breed, age
+  end
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Answer.dog = Dog.new("Cupcake", "Corgi", 5)
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
-##
-# your code here
-##
+class Person
+  attr_writer :name, :location
```

Comment:
Missed this mistake earlier... This should actually by `attr_accessor`, not `attr_writer`. `attr_writer` only gives you write access, while `attr_accessor` gives you both read _and_ write access. It's _extremely_ uncommon to use `attr_writer` - after all, how often should someone be able to change a value without being able to see what it is?